sklearn/tree/tree.py

```diff
@@ -371,6 +371,22 @@ def feature_importances_(self):
 
         return self.tree_.compute_feature_importances()
 
+    def apply(self, X):
+        """
+        Returns the index of the leaf that each sample is predicted as. 
+
+        Parameters
+        ----------
+        X: array_like, shape = (n_samples, n_features)
+        Input Samples
+
+        Returns
+        -------
+        X_leaves: array_like, shape = (n_samples,)
+        """
```

Comment:
For consistency, could you mimick the docstring of https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/ensemble/forest.py#L138 ?